src/MSALWrapper.Test/PCAWrapperTest.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using FluentAssertions;
+    using Microsoft.Authentication.MSALWrapper;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Identity.Client;
+    using Microsoft.IdentityModel.JsonWebTokens;
+    using Moq;
+    using NLog.Extensions.Logging;
+    using NLog.Targets;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// The PCA wrapper test for trying to get accounts.
+    /// </summary>
+    public class PCAWrapperTest
+    {
+        private const string TestUser = "user@microsoft.com";
+
+        private readonly IAccount userLive = new MockAccount("first@live.com");
+        private readonly IAccount userMicrosoft1 = new MockAccount("second@microsoft.com");
+        private readonly IAccount userMicrosoft2 = new MockAccount("third@microsoft.com");
+
+        private IServiceProvider serviceProvider;
+        private MemoryTarget logTarget;
+
+        // MSAL Specific Mocks
+        private Mock<IPCAWrapper> pcaWrapper;
+        private Mock<IPublicClientApplication> pcaClient;
+        private Mock<IAccount> testAccount;
+        private TokenResult tokenResult;
+
+        /// <summary>
+        /// The setup.
+        /// </summary>
+        [SetUp]
+        public void Setup()
+        {
+            // Setup in memory logging target with NLog - allows making assertions against what has been logged.
+            var loggingConfig = new NLog.Config.LoggingConfiguration();
+            this.logTarget = new MemoryTarget("memory_target");
+            loggingConfig.AddTarget(this.logTarget);
+            loggingConfig.AddRuleForAllLevels(this.logTarget);
+
+            // MSAL Mocks
+            this.testAccount = new Mock<IAccount>(MockBehavior.Strict);
+            this.testAccount.Setup(a => a.Username).Returns(TestUser);
+
+            this.pcaWrapper = new Mock<IPCAWrapper>(MockBehavior.Strict);

```

Comment:
Given this test class is testing the PCAWrapper class - we should not need any mocks of it ;)